pkg/internal/pop/msal_public.go

```diff
@@ -18,10 +18,11 @@ func AcquirePoPTokenInteractive(
 	authority,
 	clientID string,
 	options *azcore.ClientOptions,
+	isWinfieldEnv bool,
```

Comment:
This parameter is too specific to winfield here, this should be more generic such as `WithInstanceDiscovery` for example.

I also think that we should start passing optional option or struct here.